### PR TITLE
Story3 View Individual Properties.

### PR DESCRIPTION
**Pull request by turnerror**
As a: Estate Agent
I want: properties to be viewable
So that: customers can find out more info about individual properties

Requirements:
Properties in the list must link to a property detail page
Must display all available property information

### CODE REVIEW

File: property.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        <div class="stabilizerDiv"></div><!-- this empty div is required for the positioning of the h2 above-->
+    </div>
+</nav>
+<div class="container mb-4">
+    <div class="row">
+        <?php
+        echo \ArmadilloEstates\ViewHelpers\PropertyViewHelper::displayOne($property);
+        ?>
+        <div class="col-12 d-flex justify-content-center">
+            <a class="btn btn-primary" href="index.php" >Back</a>
```
Comment (bradleypurdue) [inline property.php]: Remove space before `>`.

File: property.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        </div>
+    </div>
+</div>
```
Comment (samcrudge) [inline property.php]: indentation

**Review by bradleypurdue — CHANGES_REQUESTED** [review]

File: tests/Collections/PropertyCollectionTest.php [hunk 3]
```diff
@@ -3,6 +3,9 @@
 require('vendor/autoload.php');
 
 use ArmadilloEstates\Collections\PropertyCollection;
+use ArmadilloEstates\Interfaces\IPropertyCollection;
+use ArmadilloEstates\Interfaces\IPropertyEntity;
+
```
Comment (samcrudge) [inline tests/Collections/PropertyCollectionTest.php:8]:
```suggestion
```

**Review by samcrudge — DISMISSED** [review]

**Review by bradleypurdue — DISMISSED** [review]

File: property.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<?php
+require ("vendor/autoload.php");
+$id = $_GET["id"];
```
Comment (mporam) [inline property.php]: what if there is no GET id? look at all the errors!

File: property.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+$db = \ArmadilloEstates\Database\Database::connect();
```
Comment (mporam) [inline property.php]: what if the connection fails? more errors!

File: src/Hydrators/PropertyHydrator.php [hunk 6]
```diff
@@ -28,8 +29,27 @@ public function getAllBasicProperties(): IPropertyCollection
...
+     * @return IPropertyEntity instance of the PropertyEntity returned (typehinted to IPropertyEntity)
+     **/
+    public function getCompleteListingById(string $id): IPropertyEntity {
+        $query = $this->database->prepare("
+        SELECT `description`, `agentRef`, `postcode`, `price`, `beds`, `address1`, `address2`, `town`, `typeName`, `statusName`, `properties`.`id`,  `image` FROM `properties` 
+JOIN `types` ON `properties`.`propertyType` = `types`.`typeId` 
+JOIN `status` ON `properties`.`status` = `status`.`statusId` WHERE `properties`.`id`=?;
+        ");
+        $query->execute([$id]);
+        $result = $query->setFetchMode(\PDO::FETCH_CLASS, PropertyEntity::class);
```
Comment (mporam) [inline src/Hydrators/PropertyHydrator.php]: what is result doing here??

**Review by mporam — CHANGES_REQUESTED** [review]

**Review by mporam — APPROVED** [review]

**Pull request merged**